### PR TITLE
Fix keyboard flicker and input overlap in SignInPage

### PR DESCRIPTION
**Pull request by hellohublot**
### Details

This PR fixes the problem that after clicking [`Create A New Account`, `Log In`, `Go Back`, `Back`, `Forget`], the native keyboard will flash or the input is covered

<s>This PR adds the `null value judgment` of `Hoverable.onBlur.wrapperView`, it also work fine in https://github.com/Expensify/App/pull/14507</s>

### Fixed Issues

$ https://github.com/Expensify/App/issues/16357
$ https://github.com/Expensify/App/issues/16357#issuecomment-1493072701

### Tests

Case 1:
1. Log out
2. Enter your phone number or email address and click `Continue`
3. Scroll to the bottom and click `Create A New Account`
4. Verify that the keyboard don't flicker

Case 2:
1. Log out
2. Enter your phone number or email address and click `Continue`
3. Scroll until you can hide the input (Web and desktop can be skipped)
4. Click `Go back`
5. Verify the page will automatically scroll to the top, and without covering the input

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as the other

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/20135674/236195205-7294664c-1eba-4ccb-bcba-07cf28f4cab4.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/20135674/236195239-c47be095-49a4-481f-955b-7ee2734721d7.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/20135674/236195271-96bb0144-9e7e-4609-b08a-18c02f0384c3.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/20135674/236195297-463e1301-fac0-4cdc-8481-b41b3a18c759.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/20135674/236195332-0bbb3467-4b12-423c-bfdc-815701a42ad1.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/20135674/236195352-5cde80f1-279c-4afd-a44b-e04f4fc67035.mp4

</details>

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@hellohublot I see 6 out of 5 commits say fix lint, it's better to fix them all in one commit and avoid unnecessary commits. Also you can check lint  `npm run lint` before committing changes.

**Comment by Santhosh-Sellavel** [comment]
Can you update the title meaningfully & avoid spelling mistakes?

**Comment by hellohublot** [comment]
@Santhosh-Sellavel  Thank you very much. Every time I run 'npm run lint', it gives an error of excessive memory usage. Therefore, I sometimes use the GitHub lint. I will do my best to resolve this issue. I have merged all the lint commits and also modified the PR title

File: src/components/Hoverable/index.js
Comment (Santhosh-Sellavel) [inline src/components/Hoverable/index.js:102]: Shouldn't this be `&&` ?

File: src/pages/signin/SignInPageLayout/index.js
Comment (Santhosh-Sellavel) [inline src/pages/signin/SignInPageLayout/index.js:29]: Can you move declaration top again

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Can you also update the test steps covering everything i.e every case? Also, Don't repeat the same steps in two sections. Please leave a comment saying the same as the other.

Also please resolve conflicts, thanks!

**Comment by roryabraham** [comment]
@hellohublot Looks like there are merge conflicts to resolve here.

File: src/components/Hoverable/index.js
Comment (hellohublot) [reply src/components/Hoverable/index.js:102]: The original logic is
```javascript
if (!this.wrapperView.contains(el.relatedTarget)) {
     this.setIsHovered(false)
}
```
Adding an empty judgment condition becomes
```javascript
if (this.wrapperView && !this.wrapperView.contains(el.relatedTarget)) {
     this.setIsHovered(false)
}
```
Because here ESLint needs to return in advance, so we need to change `first && second` to `!first || !second`, which will become
```javascript
if (!this.wrapperView || this.wrapperView.contains(el.relatedTarget)) {
     return;
}
```

**Comment by hellohublot** [comment]
@Santhosh-Sellavel Thanks, everything has been edited

File: src/components/Hoverable/index.js
Comment (Santhosh-Sellavel) [reply src/components/Hoverable/index.js:102]: Oops I miss read that.

File: src/components/Hoverable/index.js
Comment (Santhosh-Sellavel) [inline src/components/Hoverable/index.js:85]: Again why we are handling this here?

Comment (hellohublot) [reply src/components/Hoverable/index.js]: @Santhosh-Sellavel Sorry, just to match the below, I have removed it

File: src/components/Hoverable/index.js
Comment (Santhosh-Sellavel) [reply src/components/Hoverable/index.js:102]: how this change is also relevant to this PR?

Comment (hellohublot) [reply src/components/Hoverable/index.js:102]: > Using OR is kind of confusing, this should work right?

@ Santhosh-Sellavel
Sorry, But this doesn't work, it should be
```javascript
if (this. wrapperView && this. wrapperView. contains(el. relatedTarget)) {
     this. setIsHovered(false)
}
```
But ESLint lets us return early, so it would be the opposite


---



> how this change is also relevant to this PR?

https://github.com/Expensify/App/issues/16357#issuecomment-1504866062
I have done a full explanation here

Comment (Santhosh-Sellavel) [reply src/components/Hoverable/index.js]: @hellohublot This is how it was now without changes from this PR

```
// If wrapperView contains the target we should return here without setting false
if (this.wrapperView.contains(el.relatedTarget)) {
      return;
  }
 this.setIsHovered(false);
```

The problem is `wrapperView` could be `null` or `undefined` causing console errors. As we know only when `wrapperView` contains we should skip setting `this.setIsHovered(false)`. So the following makes sense, what's wrong here?

```js
if (this.wrapperView && this.wrapperView.contains(el.relatedTarget)) {
     return;
}
this.setIsHovered(false);
```

Comment (Santhosh-Sellavel) [reply src/components/Hoverable/index.js]: ` this.wrapperView` is `null` this makes `this.wrapperView.contains(el.relatedTarget)` check `false`, so this.setIsHovered(false) should be executed right? 

Here we skip it means like a bug to me

Comment (Santhosh-Sellavel) [reply src/components/Hoverable/index.js]: > I guess, this bug should is existed before, but everyone did not see the console error, so the error thrown by the console means that the execution is interrupted, which is consistent with our logic here

If that's the case we should avoid these changes here, maybe start reporting it on expensify-bugs on decide on it how to move on with this.

**Comment by hellohublot** [comment]
@Santhosh-Sellavel  Thanks, I will try to avoid it, sometimes after I push a `Fix Lint`, I will merge multiple commits into one, so sometimes, it is necessary to `forcefully push` to reduce and revoke `Fix Lint`, but don’t worry, this just a personal branch

**Comment by Santhosh-Sellavel** [comment]
Ensure to run the lint check comment before as I said [here](https://github.com/Expensify/App/pull/17445#issuecomment-1509253718). 

Refer to our [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=Please%20never%20force%20push%20when%20a%20PR%20review%20has%20already%20started%20(because%20this%20messes%20with%20the%20PR%20review%20history)) clealy states the same.

**Comment by hellohublot** [comment]
@Santhosh-Sellavel  Thanks, I'll try to run Lint locally

**Comment by hellohublot** [comment]
@Santhosh-Sellavel  Thanks, I have removed the `empty judgment`, because https://github.com/Expensify/App/pull/16052 is planning to remove
   `Hoverable.onBlur`, but I feel that one will `takes a long time`

If the console error can fail the QA test (but another PR has passed), we will add the empty judgment back

**Comment by Santhosh-Sellavel** [comment]
Sign-in Details not cleared 

https://user-images.githubusercontent.com/85645967/233497626-922e66ce-e2cc-4a13-bc96-1271e6961fd3.mov

But seem this is occurring in production too, behaviour change is occurring from this [PR](https://github.com/Expensify/App/pull/17119#issuecomment-1517015392)

**Comment by Santhosh-Sellavel** [comment]
@roryabraham Can we hold this PR till the above Regression is handled, because this PR is dependent on the changes made there?

**Comment by roryabraham** [comment]
Sorry, I'm not following ... can someone explain how this PR is related to https://github.com/Expensify/App/pull/16052 ?

**Comment by Santhosh-Sellavel** [comment]
> Sorry, I'm not following ... can someone explain how this PR is related to #16052 ?

It was about this change

> This PR adds the null value judgment of Hoverable.onBlur.wrapperView, it also work fine in https://github.com/Expensify/App/pull/14507

Hoverable.onBlur will be removed  #16052  or https://github.com/Expensify/App/pull/17422

**Comment by Santhosh-Sellavel** [comment]
We are waiting for this #17812

**Comment by roryabraham** [comment]
> We are waiting for this https://github.com/Expensify/App/pull/17812

Got that PR updated and ready for re-view

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/85645967/236035046-ccbe5c5c-e468-4ab3-80b7-a3124a51be65.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/85645967/236035092-7f06d04c-5a1d-4eae-bb77-edfec516e0eb.mov

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by Santhosh-Sellavel** [comment]
@hellohublot Can you pull changes from the latest main and update the test steps and screen records here as there have been some changes recently? thanks!

**Comment by Santhosh-Sellavel** [comment]
I've run into a weird issue while testing.

Unable to focus the login input after performing few steps, this happens on the main & staging as well so reported on slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1683484246745849)

https://user-images.githubusercontent.com/85645967/236695539-d31b4a3e-5837-4f19-b488-ab9fd51571e7.mp4

**Review by Santhosh-Sellavel — COMMENTED** [review]
LGTM tests well!

Found this one while testing. But happens on the main as well.
Refer this https://github.com/Expensify/App/pull/17445#issuecomment-1537512651

All yours @roryabraham!

**Comment by Santhosh-Sellavel** [comment]
@hellohublot We have conflicts here, please resolve!

**Review by Santhosh-Sellavel — APPROVED** [review]

**Comment by Santhosh-Sellavel** [comment]
> LGTM tests well!
> 
> Found this one while testing. But happens on the main as well. Refer this [#17445 (comment)](https://github.com/Expensify/App/pull/17445#issuecomment-1537512651)
> 
> All yours @roryabraham!

bump @roryabraham!

**Review by roryabraham — APPROVED** [review]

**Pull request merged**
